### PR TITLE
L40S performance enhancements

### PR DESCRIPTION
**Pull request by JCRPaquin**
Be aware, these changes were manually tested by a human but were generated by a coding agent. To the best of my knowledge & understanding, all components of this PR are reasonable changes-- I cannot vouch for the code quality or placement. The PR and all comments/responses by me are hand-written.

## Purpose

To optimize performance for 8xL40S configurations.

## Test Plan

Tested all changes with `vllm bench` and checked the top-20 logits vs cloud providers for DeepSeek-v4-Flash-0731 to confirm no/minimal changes in the top-20 distribution.

## Test Result

- 860 tok/sec continuous output at TP=8,MAX_NUM_SEQS=32
- JSD values comparable to those between Cloudflare and CoreWeave API logits
- Ad-hoc tests of model output worked well for agentic coding tasks with Oh My Pi harness

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by sakurawill** [comment]
cant it work with L40?

**Comment by JCRPaquin** [comment]
> cant it work with L40?

@sakurawill I think the code changes would work with L40, but you'd likely need to add separate quantization configs specifically for the L40 gpus.
